.circleci/config.yml

```diff
@@ -113,6 +92,115 @@ jobs:
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Determine tag name
+          command: |
+            tag=${CIRCLE_BRANCH}
+            if [ "${CIRCLE_BRANCH}" = "master" ]; then
+              tag="latest"
+            elif [[ "${CIRCLE_BRANCH}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+              tag=${CIRCLE_BRANCH:1}
+            fi
+            echo "Mapping branch ${CIRCLE_BRANCH} to ${tag}"
+            echo "export TAG=${tag}" >> ${BASH_ENV}
+      - run:
+          name: Build Docker image
+          command: |
+            docker build \
+              --no-cache \
+              -t saltyrtc/saltyrtc-server-python:${TAG} \
+              .
+      - run:
+          name: Push Docker image
+          command: |
+            docker login -u ${DOCKER_USER} -p ${DOCKER_API_KEY}
+            docker push saltyrtc/saltyrtc-server-python:${TAG}
+      - when:
+          condition: << parameters.rebuild >>
+          steps:
+            - run:
+                name: Determine supported tags
+                command: |
+                  for pattern in "<< parameters.supported-tags >>"; do
+                    tags=$(git tag -l --sort=-v:refname | grep -E $pattern | tr '\n' ' ')
+                  done
+                  echo "Building tags: ${tags}"
+                  echo "export TAGS=${tags}" >> ${BASH_ENV}
+            - run:
+                name: Build Docker images for supported tags
+                command: |
+                  function needs () {
+                    local item match="$1"
+                    shift
+                    for item; do [ "${item}" = "${match}" ] && return 1; done
+                    return 0
+                  }
+                  push_tags=()
+                  for tag in ${TAGS}; do
+                    git checkout ${tag}
+                    docker build \
+                      --no-cache \
+                      -t saltyrtc/saltyrtc-server-python:${tag:1} \
+                      .
+                    push_tags+=(${tag:1})
+                    minortag=$(echo ${tag} | sed 's/^\(v[0-9]*\.[0-9]*\)\..*$/\1/')
+                    if needs "${minortag:1}" "${push_tags[@]}"; then
+                      docker tag \
+                        saltyrtc/saltyrtc-server-python:${tag:1} \
+                        saltyrtc/saltyrtc-server-python:${minortag:1}
+                      push_tags+=(${minortag:1})
+                    fi
+                    majortag=$(echo ${tag} | sed 's/^\(v[0-9]*\)\..*$/\1/')
+                    if needs "${majortag:1}" "${push_tags[@]}"; then
+                      docker tag \
+                        saltyrtc/saltyrtc-server-python:${tag:1} \
+                        saltyrtc/saltyrtc-server-python:${majortag:1}
+                      push_tags+=(${majortag:1})
+                    fi
+                  done
+                  echo "Pushing tags: ${push_tags[@]}"
+                  echo "export PUSH_TAGS=${push_tags[@]}" >> ${BASH_ENV}
```

Comment:
Inlining this inside the config has the downside that you can't test the rebuild process locally. I guess that's fine for now, but makes debugging harder.

Reply:
Yeah and it also doesn't work because CircleCI runs sh by default. I hate shell scripting so badly.

Reply:
That's not CircleCi, it's Alpine which doesn't ship Bash by default.

Reply:
Btw, in the threema-web config I install bash and then run the rebuild script through bash :slightly_smiling_face: